node/pkg/supervisor/supervisor_test.go

```diff
@@ -0,0 +1,69 @@
+package supervisor
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func throwNil(ctx context.Context) error {
+	var x *int = nil
+	*x = 5
+
+	return nil
+}
+
+func runTest(ctx context.Context, testCase int) (result error) {
+	defer func() {
+		if r := recover(); r != nil {
+			switch x := r.(type) {
+			case string:
+				result = errors.New(x)
+			case error:
+				result = x
+			default:
+				result = errors.New("unknown panic in test1")
+			}
+		}
+	}()
+
+	errC := make(chan error)
+
+	switch testCase {
+	case 0:
+		throwNil(ctx) // fall into defer above
+	case 1:
+		RunWithScissors(ctx, errC, "test1Thread", throwNil)
+	case 2:
+		RunWithScissors(ctx, errC, "test2Thread", func(ctx context.Context) error {
+			<-ctx.Done()
+			return nil
+		})
+		throwNil(ctx)
+
+	case 3:
+		go func() { throwNil(ctx) }() // uncatchable
+	}
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-errC:
+		return err
+	}
+}
+
+func TestSupervisor(t *testing.T) {
+	for i := 0; i < 3; i++ {
+		rootCtx := context.Background()
+		ctx, fn := context.WithCancel(rootCtx)
+
+		err := runTest(ctx, i)
+
+		assert.EqualError(t, err, "runtime error: invalid memory address or nil pointer dereference")
+
+		fn()
+	}
+}
```

Comment:
```suggestion
func TestSupervisor(t *testing.T) {
	for i := 0; i < 3; i++ {
		t.Run(fmt.Sprintf("%s", i), func(t *testing.T) {
			rootCtx := context.Background()
			ctx, fn := context.WithCancel(rootCtx)
	
			err := runTest(ctx, i)
	
			assert.EqualError(t, err, "runtime error: invalid memory address or nil pointer dereference")
	
			fn()
		}
	}
}
```

By using [t.Run()](https://pkg.go.dev/testing#T.Run) you make this use [subtests](https://go.dev/blog/subtests). Subtests are nice in that you can run a specific one (say test1 or test2) and they report separately vs 3 assertions in a single unit test (if that makes sense?)